Sources/SwiftGRPCNIO/GRPCTimeout.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import NIO
+
+/// A timeout for a gRPC call.
+///
+/// Timeouts must be positive and at most 8-digits long.
+public struct GRPCTimeout: CustomStringConvertible {
+  /// A description of the timeout in the format described in the
+  /// [gRPC protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md).
+  public let description: String
+
+  private let nanoseconds: Int64
+
+  /// Creates a new GRPCTimeout with the given `amount` of the `unit`.
+  ///
+  /// `amount` must be positive and at most 8-digits.
+  private init?(_ amount: Int, _ unit: GRPCTimeoutUnit) {
+    // Timeouts must be positive and at most 8-digits.
+    guard amount >= 0, amount < 100_000_000 else { return nil }
```

Comment:
I really appreciate that you are verifying timeouts for conformance with the gRPC spec, but could we move this code to a `throw`ing factory method instead, so users know what they did wrong? Otherwise, failure to specify a valid timeout will silently result in no timeout being set.